src/Examine/LuceneEngine/Search/LuceneSearchQueryBase.cs

```diff
@@ -97,10 +97,10 @@ public IBooleanOperation Id(string id)
         public abstract IBooleanOperation ManagedQuery(string query, string[] fields = null);
         public abstract IBooleanOperation RangeQuery<T>(string[] fields, T? min, T? max, bool minInclusive = true, bool maxInclusive = true) where T : struct;
 
-        public IBooleanOperation Field(string fieldName, string fieldValue)
+        public virtual IBooleanOperation Field(string fieldName, string fieldValue)
```

Comment:
You should be able to just explicitly implement this method as IQuery.Field, since all of these methods are interfaced based, there's no reason to make anything virtual unless it wasn't part of the interface. If you explicitly implement a method on an interface, even if you are inheriting from a base class of that interface, you **are** overriding it without it being virtual.

Reply:
@Shazwazza actually if you are inheriting from LuceneSearchQueryBase it is impossible to override them from child Search Query as even if I am hiding using new keyword it is still hitting original implementation, any ideas herE?

Reply:
not the `new` keyword, you need an explicit interface implementation, like this:

```cs
public class MyQueryBase : LuceneSearchQueryBase, IQuery
    {
        //explicit implementation, this is the same as overriding
        IBooleanOperation IQuery.Field(string fieldName, string fieldValue)
        {
            //TODO: Return some value
        }
    }
```

Reply:
@Shazwazza thanks I didn't know you could override in that way, cool to know :)

Reply:
🎉 Yep, makes things a litte easier too when making your own implementations, since if it's an interface implementation you don't need to worry about making things virtual.